langlib/langlib-test/src/test/resources/test-src/arraylib_test.bal

```diff
@@ -1907,6 +1924,23 @@ function testArrayPop() {
         assertValueEquality("pop() not supported on type 'int[2]'",
         <string> checkpanic result.detail()["message"]);
     }
+
+    [int, string, int...] tuple1 = [1, "hello", 4];
+    int|string result2 = tuple1.pop();
+    assertTrue(result2 is int);
+    assertValueEquality(4, result2);
+    int|string|error result3 = trap tuple1.pop();
+    assertTrue(result3 is error);
+    if (result3 is error) {
+        assertValueEquality("{ballerina/lang.array}OperationNotSupported", result3.message());
```

Comment:
```suggestion
        assertValueEquality("{ballerina/lang.array}OperationNotSupported", result3.message());
```
Is giving `OperationNotSupported` is correct?, we support it for tuples ryt?
Imo, There should be a separate error.  ex:- Invalid operations

Reply:
It was decided in https://github.com/ballerina-platform/ballerina-lang/pull/42815#discussion_r1635091908 to use `OperationNotSupported` for `shift`. Since we are reusing `shift` for `pop` and `remove`, the error is reused.